api/client/batch_client.py

```diff
@@ -10,28 +10,33 @@
 
 from tornado import gen
 from tornado.escape import json_decode
-from tornado.httpclient import AsyncHTTPClient, HTTPRequest
+from tornado.httpclient import AsyncHTTPClient, HTTPRequest, HTTPError
 from tornado.ioloop import IOLoop
 from tornado.queues import Queue
-
+from tornado.locks import Event
 from api.client import cfg, lib, Client
 
+rate_limit_lock = Event()
+rate_limit_lock.set()
 
-class BatchClient(Client):
 
-    """API client with support for batch asynchronous queries. """
+class BatchClient(Client):
+    """API client with support for batch asynchronous queries."""
 
     _logger = None
     _http_client = None
-    
+
     def __init__(self, api_host, access_token):
         super(BatchClient, self).__init__(api_host, access_token)
         self._logger = lib.get_default_logger()
         self._http_client = AsyncHTTPClient()
 
     @gen.coroutine
     def get_data(self, url, headers, params=None):
-        """General 'make api request' function. Assigns headers and builds in retries and logging."""
+        """General 'make api request' function.
+
```

Comment:
extra line

Reply:
Extra line is from PEP 257 https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings

Was too long to be a one-line comment, so I split it up into summary and description lines